### PR TITLE
chore(flutter_counter): upgrade to Dart 2.17

### PR DESCRIPTION
**Pull request by hkuczynski**
## Status

**READY**

## Breaking Changes

NO

## Description

chore(flutter_counter): upgrade to Dart 2.17

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: examples/flutter_counter/lib/counter/view/counter_page.dart [hunk 1]
```diff
@@ -1,22 +1,21 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
-
-import '../counter.dart';
-import 'counter_view.dart';
+import 'package:flutter_counter/counter/counter.dart';
+import 'package:flutter_counter/counter/view/counter_view.dart';
```
Comment (felangel) [inline examples/flutter_counter/lib/counter/view/counter_page.dart:4]:
```suggestion
```

This should be part of the barrel file

File: examples/flutter_counter/analysis_options.yaml [hunk 2]
```diff
@@ -1,81 +1,4 @@
-analyzer:
-  strong-mode:
-    implicit-casts: false
-    implicit-dynamic: false
-  errors:
-    close_sinks: ignore
-
+include: package:very_good_analysis/analysis_options.3.0.0.yaml
```
Comment (wolfenrain) [inline examples/flutter_counter/analysis_options.yaml:1]:
```suggestion
include: package:very_good_analysis/analysis_options.3.0.1.yaml
```

**Review by wolfenrain — DISMISSED** [review]

File: examples/flutter_counter/pubspec.yaml [hunk 3]
```diff
@@ -4,32 +4,33 @@ version: 1.0.0+1
...
-    sdk: flutter
   bloc_test:
     path: ../../packages/bloc_test
-  mocktail: ^0.2.0
+  flutter_test:
+    sdk: flutter
   integration_test:
     sdk: flutter
+  mocktail: ^0.3.0
+  very_good_analysis: ^3.0.0
```
Comment (wolfenrain) [inline examples/flutter_counter/pubspec.yaml:33]:
```suggestion
  very_good_analysis: ^3.0.1
```

File: examples/flutter_counter/lib/counter/view/counter_page.dart (hunk 1 above)
Comment (hkuczynski) [reply examples/flutter_counter/lib/counter/view/counter_page.dart]: Fixed

**Review by felangel — APPROVED** [review]

**Pull request merged**